### PR TITLE
Google Photos (Independent Publisher)

### PR DESCRIPTION
**Pull request by juliamuiruri4**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by juliamuiruri4** [comment]
first-five-operations-Test-Success

**Comment by juliamuiruri4** [comment]
Validation-Test-Successful

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,161 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "1.0.0",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:4]: @juliamuiruri4 please change the version to "1.0"

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        "operationId": "ListAlbums",
+        "parameters": [],
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:30]: @juliamuiruri4 default responses is not allowed. Please define a proper response, please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/coding-standards#responses)

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            "name": "body",
+            "in": "body",
+            "schema": {
+              "type": "string"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:51]: default responses is not allowed. Please define a proper response, please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/coding-standards#responses)

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        "operationId": "ListItems",
+        "parameters": [],
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:72]: default responses is not allowed. Please define a proper response, please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/coding-standards#responses)

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            "default": "<string>",
+            "in": "query",
+            "type": "string",
+            "x-ms-summary": "Create Item",
+            "description": "Create Media Item - Photo, video etc",
+            "required": true
+          }
+        ],
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:99]: @juliamuiruri4 default responses is not allowed. Please define a proper response, please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/coding-standards#responses)

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        "operationId": "ListSharedAlbums",
+        "parameters": [],
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:114]: @juliamuiruri4 default responses is not allowed. Please define a proper response, please see [here](https://learn.microsoft.com/en-gb/connectors/custom-connectors/coding-standards#responses)

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        "operationId": "ListAlbums",
+        "parameters": [],
+        "responses": {
+          "default": {
+            "description": "200",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:32]: @juliamuiruri4 empty schema is not allowed please define a proper schema.

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+            "schema": {
+              "type": "string"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:53]: @juliamuiruri4 empty schema is not allowed please define a proper schema.

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+        "operationId": "ListItems",
+        "parameters": [],
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:74]: @juliamuiruri4 empty schema is not allowed please define a proper schema.

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+            "type": "string",
+            "x-ms-summary": "Create Item",
+            "description": "Create Media Item - Photo, video etc",
+            "required": true
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:101]: @juliamuiruri4 empty schema is not allowed please define a proper schema.

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+        "operationId": "ListSharedAlbums",
+        "parameters": [],
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:116]: @juliamuiruri4 empty schema is not allowed please define a proper schema.

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+        ],
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        }
+      }
+    },
+    "/v1/albums/%7B%7BalbumId%7D%7D": {},
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:58]: @juliamuiruri4 can you please explain why you have an empty paths? May I ask you please to remove all empty paths from your swagger file.

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+        "parameters": [],
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        }
+      }
+    },
+    "/v1/mediaItems/%7B%7BmediaItemId%7D%7D": {},
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:79]: @juliamuiruri4 can you please explain why you have an empty paths? May I ask you please to remove all empty paths from your swagger file.

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+        ],
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        }
+      }
+    },
+    "/v1/mediaItems:batchCreate": {},
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:106]: @juliamuiruri4 can you please explain why you have an empty paths? May I ask you please to remove all empty paths from your swagger file.

File: independent-publisher-connectors/Google Photos/apiProperties.json [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+            "refreshUrl": {
+              "value": "https://accounts.google.com/o/oauth2/token"
+            }
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#da3b01",
+    "capabilities": [],
+    "publisher": "Julia Muiruri"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiProperties.json:34]: @juliamuiruri4 please add the "stackOwner" property, you can follow this template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json).
Please add "stackOwner": "Name of company or organization that connector is connecting to."

File: independent-publisher-connectors/Google Photos/readme.md [hunk 16]
```diff
@@ -0,0 +1,36 @@
+# Google Photos
+Integrate smart, easy-to-use photo and video features into your product and reach hundreds of millions of people who choose Google Photos.
+
+<br>
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/readme.md:4]: @juliamuiruri4 please remove all <br> from the readme file.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
+            "description": "200",
+            "schema": {}
+          }
+        }
+      },
+      "post": {
+        "summary": "Create New Album",
+        "description": "Creates an album in a user's Google Photos library.\n\n### HTTP request\n\n`POST https://photoslibrary.googleapis.com/v1/albums`\n\nThe URL uses [gRPC Transcoding](https://github.com/googleapis/googleapis/blob/master/google/api/http.proto) syntax.\n\n### Request body\n\nThe request body contains data with the following structure:\n\nJSON representation\n\n```json\n{\n  \"album\": {\n    object (Album)\n  }\n}\n```\n\n|Fields|||\n|----|----|----|\n|`album`|`object` ([Album](https://developers.google.com/photos/library/reference/rest/v1/albums#Album))|The album to be created.|\n\n### Response body\n\nIf successful, the response body contains a newly created instance of [Album](https://developers.google.com/photos/library/reference/rest/v1/albums#Album).\n\n### Authorization Scopes\n\nRequires one of the following OAuth scopes:\n\n*   `https://www.googleapis.com/auth/photoslibrary`\n*   `https://www.googleapis.com/auth/photoslibrary.appendonly`\n*   `https://www.googleapis.com/auth/photoslibrary.sharing`",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:38]: @juliamuiruri4 the description will show up as the operation description when selecting the information button. We suggest keeping it short to fit in the text box. No period required if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 18]
```diff
@@ -0,0 +1,161 @@
...
...est returns the rows after the `pageToken`. The `pageToken` should be the value returned in the `nextPageToken` parameter in the response to the `listMediaItems` request.|\n\n### Request body\n\nThe request body must be empty.\n\n### Response body\n\nIf successful, the response body contains data with the following structure:\n\nList of all media items from the user's Google Photos library.\n\nJSON representation\n\n```json\n{\n  \"mediaItems\": [\n    {\n      object (MediaItem)\n    }\n  ],\n  \"nextPageToken\": string\n}\n```\n\n|Fields|||\n|-|-|-\n|`mediaItems[]`|`object` ([MediaItem](https://developers.google.com/photos/library/reference/rest/v1/mediaItems#MediaItem))|Output only. List of media items in the user's library.\n|`nextPageToken`|`string`|Output only. Token to use to get the next set of media items. Its presence is the only reliable indicator of more media items being available in the next request.\n\n### Authorization Scopes\n\nRequires one of the following OAuth scopes:\n\n*   `https://www.googleapis.com/auth/photoslibrary`\n*   `https://www.googleapis.com/auth/photoslibrary.readonly`\n*   `https://www.googleapis.com/auth/photoslibrary.readonly.appcreateddata`\n",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:68]: @juliamuiruri4 the description will show up as the operation description when selecting the information button. We suggest keeping it short to fit in the text box. No period required if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 19]
```diff
@@ -0,0 +1,161 @@
...
...evelopers.google.com/photos/library/reference/rest/v1/mediaItems/batchGet#MediaItemResult.FIELDS.status).\n\n### Authorization Scopes\n\nRequires one of the following OAuth scopes:\n\n*   `https://www.googleapis.com/auth/photoslibrary`\n*   `https://www.googleapis.com/auth/photoslibrary.readonly`\n*   `https://www.googleapis.com/auth/photoslibrary.readonly.appcreateddata`\n\nMediaItemResult\n---------------\n\nResult of retrieving a media item.\n\nJSON representation\n\n```json\n{\n  \"status\": {\n    object (Status)\n  },\n  \"mediaItem\": {\n    object (MediaItem)\n  }\n}\n```\n\n|Fields|||\n|-|-|-\n|`status`|`object` ([Status](https://developers.google.com/photos/library/reference/rest/v1/Status))|If an error occurred while accessing this media item, this field is populated with information related to the error. For details regarding this field, see [Status](https://developers.google.com/photos/library/reference/rest/v1/Status).\n|`mediaItem`|`object` ([MediaItem](https://developers.google.com/photos/library/reference/rest/v1/mediaItems#MediaItem))|Media item retrieved from the user's library. It's populated if no errors occurred and the media item was fetched successfully.\n",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:85]: @juliamuiruri4 the description will show up as the operation description when selecting the information button. We suggest keeping it short to fit in the text box. No period required if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json [hunk 20]
```diff
@@ -0,0 +1,161 @@
...
...redAlbums` request.\n|`excludeNonAppCreatedData`|`boolean`|If set, the results exclude media items that were not created by this app. Defaults to false (all albums are returned). This field is ignored if the photoslibrary.readonly.appcreateddata scope is used.\n\n### Request body\n\nThe request body must be empty.\n\n### Response body\n\nIf successful, the response body contains data with the following structure:\n\nList of shared albums requested.\n\nJSON representation\n\n```json\n{\n  \"sharedAlbums\": [\n    {\n      object (Album)\n    }\n  ],\n  \"nextPageToken\": string\n}\n```\n\n|Fields|||\n|-|-|-\n|`sharedAlbums[]`|`object` ([Album](https://developers.google.com/photos/library/reference/rest/v1/albums#Album))|Output only. List of shared albums.\n|`nextPageToken`|`string`|Output only. Token to use to get the next set of shared albums. Populated if there are more shared albums to retrieve for this request.\n\n### Authorization Scopes\n\nRequires one of the following OAuth scopes:\n\n*   `https://www.googleapis.com/auth/photoslibrary`\n*   `https://www.googleapis.com/auth/photoslibrary.readonly`\n*   `https://www.googleapis.com/auth/photoslibrary.readonly.appcreateddata`\n",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Google Photos/apiDefinition.swagger.json:110]: @juliamuiruri4 the description will show up as the operation description when selecting the information button. We suggest keeping it short to fit in the text box. No period required if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

**Review by Amjed-Ayoub — DISMISSED** [review]

File: independent-publisher-connectors/Google Photos/apiDefinition.swagger.json (hunk 1 above)
Comment (juliamuiruri4) [reply independent-publisher-connectors/Google Photos/apiDefinition.swagger.json]: Done

**Comment by juliamuiruri4** [comment]
@Amjed-Ayoub I've responded to your requested changes. Please advise on what action to take next.
Regards

**Comment by Amjed-Ayoub** [comment]
Hello @juliamuiruri4, thank you so much for fixing your connector artifacts. Please remove the Proposal from your pull request title, so we can move forward with the certification process.
Thank you so much again.

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @juliamuiruri4,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by juliamuiruri4** [comment]
Thank you so much @Amjed-Ayoub for sharing this update. Looking forward to it.

**Comment by Amjed-Ayoub** [comment]
Hello @juliamuiruri4,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Review by sriyen-msft — APPROVED** [review]

**Comment by sriyen-msft** [comment]
@juliamuiruri4 Congratulations! The deployment of your connector and documentation page is now complete, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program, Thanks.

**Pull request merged**